libtsuba/src/RDGCore.cpp

```diff
@@ -131,6 +132,40 @@ schemify(const std::vector<katana::PropStorageInfo>& prop_info_list) {
   }
   return arrow::schema(fields);
 }
+
+template <typename StorageType>
+katana::Result<katana::NUMAArray<katana::EntityTypeID>>
+LoadAndMaybeConvertTypeIDArray(
+    size_t begin, size_t end, const katana::Uri& types_path,
+    const katana::RDGPartHeader& part_header) {
+  katana::NUMAArray<katana::EntityTypeID> types;
+
+  // NB: we add sizeof(EntityTypeIDArrayHeader) to every range element because
+  // the structure of this file is
+  // [header, value, value, value, ...]
+  // Recent storage formats remove this header
+  size_t header_size = part_header.unstable_storage_format()
+                           ? 0
+                           : sizeof(katana::EntityTypeIDArrayHeader);
+  size_t storage_begin = begin * sizeof(StorageType) + header_size;
+  size_t storage_end = end * sizeof(StorageType) + header_size;
+
+  katana::FileView fv;
+
+  KATANA_CHECKED_CONTEXT(
+      fv.Bind(types_path.string(), storage_begin, storage_end, true),
+      "loading node type id array, begin: {}, end: {}", begin, end);
+
+  types.allocateInterleaved(end - begin);
```

Comment:
Do we expect a new copy being allocated everytime we call the API? Previously, `Bind` would mmap a file in and the user code would decide to copy into a `NUMAArray` or whatever.

Reply:
Currently many users make this copy anyway. Users that aren't making this copy can still use the original interface. Reducing the number of copies and/or using a NUMA-aware allocator for the original `Bind()` are left as future work.

Reply:
OK. This makes sense given the old interface is still there.